hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/ContractsModule.java

```diff
@@ -210,6 +231,15 @@ static ContractCreationProcessor provideV_0_34ContractCreateProcessor(
         return new ContractCreationProcessor(gasCalculator, evm, true, List.copyOf(validationRules), 1);
     }
 
+    @Provides
+    @Singleton
+    @IntoMap
+    @StringKey(ContractsV_0_38Module.EVM_VERSION_0_38)
+    static ContractCreationProcessor provideV_0_38ContractCreateProcessor(
```

Comment:
Q: So even when the details are exactly the same we still have to add a new method for each new version?
Is that because of the lookup logic based on the key above?

Reply:
This is how it needs to work to support dagger.  Because we inject a different version of the EVM (marked with the @V_0_38 annotation) we need a whole new provides definition.